### PR TITLE
refactor(SPV-1232): admin endpoints group - Transactions API.

### PR DESCRIPTION
**Pull request by mgosek-4chain**
**Major Changes:**

**Admin API Enhancements:**

- Added support for transactions Admin API endpoints, including comprehensive documentation.
- Standardized expected JSON file names in the user transactions API package for consistency.
- Updated test case names and exported the transaction filter builder to enable reuse in the Admin API package.
- Performed minor cleanups: corrected typos and added missing "_given-when_" sections in test cases.

# Pull Request Checklist

- [X] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-go-client/blob/main/.github/CODE_STANDARDS.md)
- [X] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-go-client/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] 🏠 I tested my changes locally.
- [X] ✅ I have provided tests for my changes.
- [X] 📝 I have used conventional commits.
- [X] 📗 I have updated any related documentation.
- [X] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: internal/api/v1/admin/transactions/transactions_api.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		SetQueryParams(params.ParseToMap()).
+		Get(a.url.String())
+	if err != nil {
+		return nil, fmt.Errorf("HTTP response failure: %w", err)
+	}
+
+	return &result, nil
+}
+
+func NewAPI(url *url.URL, httpClient *resty.Client) *API {
```
Comment (ac4ch) [inline internal/api/v1/admin/transactions/transactions_api.go:68]: consider to check this per nil, like i suggested in other PR

File: internal/api/v1/admin/transactions/transactions_api.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	return &API{url: url.JoinPath(route), httpClient: httpClient}
+}
+
+func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
```
Comment (ac4ch) [inline internal/api/v1/admin/transactions/transactions_api.go:72]: same why not used in Transactions

**Review by dzolt-4chain — APPROVED** [review]

**Review by chris-4chain — APPROVED** [review]

File: internal/api/v1/admin/transactions/transactions_api.go (hunk 1 above)
Comment (mgosek-4chain) [reply internal/api/v1/admin/transactions/transactions_api.go:68]: Added checks to initUserAPI, initAdminAPI functions. The design intent was always to pass dependencies to lower layers in the correct state (i.e from _spvwallet -> transactions_ etc.).

File: internal/api/v1/admin/transactions/transactions_api.go (hunk 2 above)
Comment (mgosek-4chain) [reply internal/api/v1/admin/transactions/transactions_api.go:72]: I will handle such cases in the separate PR.

**Pull request merged**